src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -493,6 +509,24 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
     return this._getConnectedElement().nativeElement.getBoundingClientRect().width;
   }
 
+  /** Returns a stream of click events from outside of the autocomplete panel. */
+  private _getOutsideClickStream(document: Document): Observable<any> {
+    return document ?
+      RxChain.from(merge(
+        fromEvent(document, 'click'),
+        fromEvent(document, 'touchend')
+      )).call(filter, (event: MouseEvent | TouchEvent) => {
+        const clickTarget = event.target as HTMLElement;
+        const formField = this._formField ?
+          this._formField._elementRef.nativeElement : null;
+
+        return this._panelOpen &&
+          clickTarget !== this._element.nativeElement &&
+          (!formField || !formField.contains(clickTarget)) &&
+          (!!this._overlayRef && !this._overlayRef.overlayElement.contains(clickTarget));
+      }).call(share).result() : observableOf();
```

Comment:
Note that the `observableOf()` here isn't the same as the `observableOf(null)` that we had before. `observableOf()` returns an observable that'll never emit, whereas `observableOf(null)` will emit immediately. In this particular case it shouldn't matter since it's a fallback for Universal.

Reply:
Yes aware of the difference. It was intentional to use observableOf() rather than observableOf(null), since it observableOf(null) would emit when there hasn't been a click.